macros/src/types/enum.rs

```diff
@@ -124,9 +122,13 @@ fn format_variant(
                         (Some(_), Some(_)) => syn_err!("`type` is not compatible with `as`"),
                         (Some(type_override), None) => quote! { #type_override },
                         (None, Some(type_as)) => {
-                            format_type(&syn::parse_str::<Type>(&type_as)?, dependencies, generics)
+                            let ty = syn::parse_str::<Type>(&type_as)?;
+                            quote!(<#ty as ts_rs::TS>::name())
```

Comment:
This is just one example of how every use of `generics::format_type` got replaced with `TS::name`.